.github/workflows/undesirable-test-additions.yaml

```diff
@@ -0,0 +1,106 @@
+name: Check for Undesirable Code
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  check_undesirable_code:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Fetch all history for comparison
```

Comment:
add blob filter, so we don't need all blobs

Reply:
It now calls `/usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules --filter=blob:none origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/*`